### PR TITLE
Upgrade todomvc-nix into flake features.

### PR DESCRIPTION
**Pull request by Rizary**
This is my first attempt to refactor all `todomvc-nix` code to use the flake feature. It's based on this #10 as a starter.

# What changed?
1. Adding flake style development into the project
2. I add `.devcontainer` and `.vscode/setting.json` so that VSCode's user can use `remote-container` plugin when playing with this project
3. I add direnv `.envrc` to make direnv work in the project
4. Clean up `README.md` and add `Docs` folder to narrow each architecture explanation.
5. Group the project example _per_ language for clarity and easy project management
6. Adding full-stack Haskell and Rust todomvc
7. Delete all CI/CD related scripts and `.nix` file for future replacement (GitHub Actions in mind)

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -1,21 +1,108 @@
-let
-  pkgs = import ./nix {};
-in
+{ pkgs }:
```
Comment (zimbatm) [inline shell.nix:1]: this breaks when running `nix-shell` because there is no default value for `pkgs`.

File: shell.nix [hunk 2]
```diff
@@ -1,21 +1,108 @@
...
 
 with pkgs;
 
-mkShell {
-  inputsFrom = [
-    frontend
-    backend
+mkDevShell {
```
Comment (zimbatm) [inline shell.nix:5]: this looks like a duplicate of `devshell.toml`?

Comment (Rizary) [reply shell.nix:5]: Yeah, but it's more complicated. I left `devshell.toml` as-is for example purpose.

File: shell.nix (hunk 1 above)
Comment (Rizary) [reply shell.nix:1]: The best way to do this is similar to what you're doing in https://github.com/nix-community/nixpkgs-fmt/blob/master/shell.nix, but I still need to create `mkshell.nix` and import it here: https://github.com/nix-community/nixpkgs-fmt/blob/master/flake.nix#L34

I don't know, should we retain the `non-flake-enabled Nix versions`, given that we want to show Nix user how to use `flake`?

File: flake.lock [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      },
+      "original": {
+        "owner": "mozilla",
+        "repo": "nixpkgs-mozilla",
+        "type": "github"
+      }
+    },
+    "naersk": {
+      "inputs": {
+        "nixpkgs": "nixpkgs"
```
[the remark this answers is not in the document]
Comment (Rizary) [reply flake.lock:86]: noted.

File: shell.nix (hunk 1 above)
Comment (Rizary) [reply shell.nix:1]: I have update `default.nix` and `shell.nix`

**Pull request merged**